Readme.md

```diff
@@ -1,32 +1,138 @@
-# Hello Wise Wizard. Welcome to the magic land of DW.
+# Hello Wise Wizard. Welcome to the magic land of DataWeave.
 
-This Grimore contains a collection of interesting DataWeave spells that can be executed with [our CLI](https://github.com/mulesoft-labs/data-weave-native).
+This grimoire contains a collection of interesting DataWeave spells that can be executed with our [DataWeave CLI](https://github.com/mulesoft-labs/data-weave-native).
 
-You can create your own grimore and be part of the DataWeave Wizards Secret Club.
+You can create your own grimoire and be part of the DataWeave Wizards Secret Club.
 
-## How to create a SPELL
+Let's start defining some concepts:
 
-- wizard-name = Github User Name
+## Wizard
+A `wizard` is just a name that we trust and we want to have all their `spells` available.
 
-All you need is 
- - A repo `*<wizard-name>*-data-weave-grimoire`
- - A folder that is going to be your spell name i.e HelloWorld/
- - Then a file Main.dwl that needs to be under <spell_name>/src/Main.dwl in our case HelloWorld/src/Main.dwl
- - And then write your script inside Main.dwl for example:
+A `wizard` has a `grimoire`, a GitHub repository that is inferred by `https://github.com/${wizard_name}/${wizard_name}-data-weave-grimoire`, that includes all the `spells` from this `wizard`.
+
+## Grimoire
+
+A `grimoire` is a collection of `spells` from a given `wizard`. 
```

Comment:
A `Grimoire` is a collection of `spell` created by a `wizard`. The grimoire is hosted in a GitHub repo in the default branch. Currently there is no support for versioning or branching scheme.